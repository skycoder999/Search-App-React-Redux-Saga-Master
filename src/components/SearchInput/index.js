import React, { useState, useEffect } from "react";
import searchIcon from "../../assets/images/search-icon.png";
import { TEXT_SEARCH_PLACEHOLDER } from "../../utils/constants";

const SearchInput = ({ selected = "", onChange }, inputRef) => {
  const [isFocus, setIsFocus] = useState(false);
  const [value, setValue] = useState("");

  const handleChange = () => {
    setValue(inputRef.current.value);
    onChange(inputRef.current.value);
  };

  const onFocus = () => {
    setIsFocus(true);
  };

  const onBlur = () => {
    setIsFocus(false);
  };

  const clearInput = () => {
    inputRef.current.value = "";
    inputRef.current.focus();
    handleChange();
  };

  useEffect(() => {
    if (selected) {
      inputRef.current.value = selected;
    }
  }, [selected, inputRef]);

  return (
    <div className={"suggest-input" + (isFocus ? " focused" : "")}>
      <input ref={inputRef} placeholder={TEXT_SEARCH_PLACEHOLDER} type="text" onBlur={onBlur} onFocus={onFocus} onChange={handleChange} />
      {value ? (
        <span className="suggest-input__clear" onClick={clearInput}>
          X
        </span>
      ) : null}
      {!value ? <img className="suggest-input__icon" src={searchIcon} alt="" /> : null}
    </div>
  );
};

export default React.forwardRef(SearchInput);
