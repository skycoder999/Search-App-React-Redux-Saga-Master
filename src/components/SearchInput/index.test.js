import SearchInput from "./index";

let wrapper, consoleError;
let handleChangeSpy = jest.fn();

beforeEach(() => {
  wrapper = shallow(<SearchInput selected="" />);
  consoleError = console.error;
  console.error = jest.spyOn(console, "error");
});

afterEach(() => {
  console.error = consoleError;
});

describe("SearchInput Component", () => {
  it("renders without crashing", () => {
    expect(wrapper.exists()).toBe(true);
  });

  it("matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
