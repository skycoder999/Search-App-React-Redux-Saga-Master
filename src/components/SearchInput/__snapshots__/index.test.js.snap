// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput Component matches the snapshot 1`] = `ShallowWrapper {}`;
